lib/tasks/offers.rake

```diff
@@ -0,0 +1,19 @@
+# frozen_string_literal: true
+
+namespace :offers do
+  desc 'Lapse all offers that were sent before a certain date'
+  task :lapse_sent, [:date] => :environment do |_task, args|
+    raise 'Please supply a date until which sent offers should be considered to lapse!' unless args[:date]
+
+    datetime = args[:date].to_datetime
+    Offer.where("state = ? AND (sent_at < ? OR sent_at IS NULL)", "sent", datetime).update_all(state: "lapsed")
```

Comment:
Actually, Rubocop gives a warning about `update_all` skipping validations. It might be worth it to look at ActiveRecord Batch operations (https://api.rubyonrails.org/v5.1/classes/ActiveRecord/Batches.html) and do something like:

```
Offer.where("state = ? AND (sent_at < ? OR sent_at IS NULL)", "sent", datetime).in_batches do |offer|
  relation.update(state: "lapsed")
end
```